pkg/controller/nodenetworkconfigurationpolicy/selectors/selectors.go

```diff
@@ -0,0 +1,25 @@
+package selectors
+
+import (
+	"github.com/go-logr/logr"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+type Selectors struct {
+	client client.Client
+	policy nmstatev1alpha1.NodeNetworkConfigurationPolicy
+	logger logr.Logger
+}
+
+func NewFromPolicy(client client.Client, policy nmstatev1alpha1.NodeNetworkConfigurationPolicy) Selectors {
+	selectors := Selectors{
+		client: client,
+		policy: policy,
```

Comment:
If we introduce `NewFromSomethingElse`, this would not work again. The `Selectors` object should have an attribute `nodeSelectors` and this attribute should be populated in this constructor.

Reply:
We are not going to do NewFromSomethingElse

Reply:
I want to limit the structure to minimum that is needed. Small scopes are easier to read and debug. But I won't block this PR anymore, this is just a nit a maybe a matter of taste.